### PR TITLE
Exp/refactor scala version

### PR DESCRIPTION
**Pull request by davidB**
refactor the code:
- remove duplicate in dependencies and classpath build
- remove some "if scala.version.major == 3" (all except one)

Strange behavior to be able to compile scala 3 (in incremental mode), scala-library 2 should be provide, it's seems to be a transitive dependencies of the compiler, from debug print (I'll remove them)

```txt
[INFO] >>> libraryJars from sc: /home/david/.m2/repository/org/scala-lang/scala3-library_3.0.0-RC2/3.0.0-RC2/scala3-library_3.0.0-RC2-3.0.0-RC2.jar
[INFO] >>> libraryJars from finder compiler: /home/david/.m2/repository/org/scala-lang/scala3-library_3.0.0-RC2/3.0.0-RC2/scala3-library_3.0.0-RC2-3.0.0-RC2.jar - /home/david/.m2/repository/org/scala-lang/scala-library/2.13.5/scala-library-2.13.5.jar
[INFO] >>> compilerJars from finder compiler: /home/david/.m2/repository/org/scala-lang/scala3-compiler_3.0.0-RC2/3.0.0-RC2/scala3-compiler_3.0.0-RC2-3.0.0-RC2.jar - /home/david/.m2/repository/org/scala-lang/scala3-interfaces/3.0.0-RC2/scala3-interfaces-3.0.0-RC2.jar - /home/david/.m2/repository/org/scala-lang/scala3-library_3.0.0-RC2/3.0.0-RC2/scala3-library_3.0.0-RC2-3.0.0-RC2.jar - /home/david/.m2/repository/org/scala-lang/scala-library/2.13.5/scala-library-2.13.5.jar - /home/david/.m2/repository/org/scala-lang/tasty-core_3.0.0-RC2/3.0.0-RC2/tasty-core_3.0.0-RC2-3.0.0-RC2.jar - /home/david/.m2/repository/com/novocode/junit-interface/0.11/junit-interface-0.11.jar - /home/david/.m2/repository/junit/junit/4.11/junit-4.11.jar - /home/david/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar - /home/david/.m2/repository/org/scala-sbt/test-interface/1.0/test-interface-1.0.jar - /home/david/.m2/repository/org/scala-lang/modules/scala-asm/9.1.0-scala-1/scala-asm-9.1.0-scala-1.jar - /home/david/.m2/repository/org/scala-sbt/compiler-interface/1.3.5/compiler-interface-1.3.5.jar - /home/david/.m2/repository/com/google/protobuf/protobuf-java/3.7.0/protobuf-java-3.7.0.jar - /home/david/.m2/repository/org/scala-sbt/util-interface/1.3.0/util-interface-1.3.0.jar - /home/david/.m2/repository/org/jline/jline-reader/3.19.0/jline-reader-3.19.0.jar - /home/david/.m2/repository/org/jline/jline-terminal/3.19.0/jline-terminal-3.19.0.jar - /home/david/.m2/repository/org/jline/jline-terminal-jna/3.19.0/jline-terminal-jna-3.19.0.jar - /home/david/.m2/repository/net/java/dev/jna/jna/5.3.1/jna-5.3.1.jar
```

When I used only "libraryJars from sc" (aka only scala3-library) the incremental compilation failed.

### CODE REVIEW

File: src/main/java/scala_maven_dependency/ArtifactIds4Scala3.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package scala_maven_dependency;
+
+import java.util.Arrays;
+import java.util.List;
+import scala_maven.VersionNumber;
+
+public class ArtifactIds4Scala3 implements ArtifactIds {
+  protected static final String SCALA_LIBRARY_ARTIFACTID = "scala3-library";
+  protected static final String SCALA_INTERFACES_ARTIFACTID = "scala3-interfaces";
+  protected static final String SCALA_REFLECT_ARTIFACTID = "scala3-reflect";
```
Comment (rajcspsg) [inline src/main/java/scala_maven_dependency/ArtifactIds4Scala3.java:26]: scala3 doesn't have reflect jar. may be we should remove this.

Comment (davidB) [reply src/main/java/scala_maven_dependency/ArtifactIds4Scala3.java]: thanks, I refactored and was able to also remove it from scala2

File: src/main/java/scala_maven/ScalaCompilerLoader.java [hunk 2]
```diff
@@ -19,10 +19,10 @@
 import java.net.URL;
 import java.net.URLClassLoader;
 
-public class Scala3CompilerLoader extends URLClassLoader {
+public class ScalaCompilerLoader extends URLClassLoader {
```
Comment (adpi2) [inline src/main/java/scala_maven/ScalaCompilerLoader.java:22]: This class is, in large part, copied from the [lampefl/dotty](https://github.com/lampepfl/dotty/blob/96401b68a8bce4c125859806b0c24a1ffe3cbc1e/sbt-bridge/src/xsbt/CompilerClassLoader.java) code base wich is licensed under [Apache License 2.0](https://github.com/lampepfl/dotty/blob/master/LICENSE). So I think the "Dotty (https://dotty.epfl.ch) Copyright 2012-2020 EPFL Copyright 2012-2020 Lightbend, Inc" should be preserved.

Comment (davidB) [reply src/main/java/scala_maven/ScalaCompilerLoader.java:24]: Thanks, I updated the header. It allow to see that the header was set by the formatter (I didn't configure it) and that the wrong license header is injected. It should be unlicenced. I'll do a dedicated commit (all file to update)

**Pull request merged**